release-calendar/src/client/components/Calendar.tsx

```diff
@@ -47,37 +44,50 @@ export class Calendar extends React.Component<CalendarProps, CalendarState> {
   constructor(props: Readonly<CalendarProps>) {
     super(props);
     this.state = {
-      allEvents: new Map<Channel, EventSourceInput>(),
+      allEvents: new Map<Channel, ReleaseEventInput[]>(),
       singleEvents: [],
     };
     this.apiService = new ApiService();
   }
 
   async componentDidMount(): Promise<void> {
     const promotions = await this.apiService.getPromotions();
-    this.setState({allEvents: getAllPromotionEvents(promotions)});
+    this.setState({allEvents: getAllEvents(promotions)});
   }
 
   async componentDidUpdate(prevProps: CalendarProps): Promise<void> {
     if (prevProps.singleRelease != this.props.singleRelease) {
       if (this.props.singleRelease != null) {
-        const release = await this.apiService.getRelease(
+        const promotionsOfSingleRelease = await this.apiService.getSinglePromotions(
           this.props.singleRelease,
         );
-        this.setState({singleEvents: getSingleReleaseEvents(release)});
+        this.setState((prevState: CalendarState) => ({
+          singleEvents: getSingleReleaseEvents(
+            promotionsOfSingleRelease,
+            prevState.allEvents,
+          ),
+        }));
       } else {
         this.setState({singleEvents: []});
       }
     }
   }
 
   render(): JSX.Element {
-    const displayEvents: EventSourceInput[] =
-      this.props.singleRelease != null
-        ? this.state.singleEvents
-        : this.props.channels
-            .filter((channel) => this.state.allEvents.has(channel))
-            .map((channel) => this.state.allEvents.get(channel));
+    let displayEvents: ReleaseEventInput[] = [];
+    if (this.props.singleRelease != null) {
+      displayEvents = this.state.singleEvents;
+    } else {
+      this.props.channels
+        .filter((channel) => this.state.allEvents.has(channel))
+        .forEach(
+          (channel) =>
+            (displayEvents = [
+              ...displayEvents,
+              ...this.state.allEvents.get(channel),
+            ]),
```

Comment:
When creating an arrow with a single expression not inside a block (e.g., `() => expr` instead of `() => { expr; }`) it's equivalent to having a block  that returns that single expression. That is...
```ts
// this:
(channel) => someFoo(channel);

// is basically this:
(channel) => {
  return someFoo(channel);
}
```

This means that the arrow function here is equivalent to:
```ts
(channel) => {
  return (displayEvents = [
    ...displayEvents,
    ...this.state.allEvents.get(channel),
  ]);
}
```

and that's a little confusing, because it makes the reader wonder why you're returning an assignment expression ("`return (x = y)`") to a function that doesn't take arguments (`forEach` doesn't act on return values from its callback function)

In this case, it makes better sense to do:
```ts
(channel) => {
  displayEvents = [
    ...displayEvents,
    ...this.state.allEvents.get(channel),
  ];
}
```

Without a return statement. An even more concise version would be:
```ts
(channel) => {
  displayEvents.push(...this.state.allEvents.get(channel));
}
```

Reply:
Great comment!

Reply:
Yes! Thanks, for explaining :)